src/ctapipe/monitoring/outlier.py

```diff
@@ -0,0 +1,131 @@
+        outlier detection approach. The function examines the statistics in the
+        given column of the table and returns a boolean mask indicating which
+        entries are considered as outliers.
+
+        Parameters
+        ----------
+        column : astropy.table.Column
+            column with chunk-wise aggregated statistic values (mean, median, or std)
+            of shape (n_entries, n_channels, n_pix)
+
+        Returns
+        -------
+        boolean mask
+            mask of outliers of shape (n_entries, n_channels, n_pix)
+        """
+        pass
+
+
+class RangeOutlierDetector(OutlierDetector):
+    """
+    Detect outliers based on a valid range.
+
+    The clipping interval to set the thresholds for detecting outliers corresponds to
+    a configurable range of valid statistic values.
+    """
+
+    validity_range = List(
+        [1.0, 2.0],
+        help=(
+            "Range of valid statistic values (in units of the image value)."
+            "Values outside the range are identified as outliers."
+        ),
+    ).tag(config=True)
+
+    def __call__(self, column):
+        # Remove outliers is statistical values out a given range
+        outliers = np.logical_or(
+            column < self.validity_range[0],
+            column > self.validity_range[1],
+        )
+        return outliers
+
+
+class MedianOutlierDetector(OutlierDetector):
+    """
+    Detect outliers based on the deviation from the camera median.
+
+    The clipping interval to set the thresholds for detecting outliers is computed by multiplying
+    the configurable factors and the camera median of the statistic values.
+    """
+
+    median_range_factors = List(
+        [-1.0, 1.0],
+        help=(
+            "Multiplicative factors (unitless) applied to the camera median"
+            "of the provided statistic values to define a valid range based on the"
+            "deviation of the values to its camera median."
+        ),
+    ).tag(config=True)
+
+    def __call__(self, column):
+        # Camera median
+        camera_median = np.ma.median(column, axis=2)
+        # Detect outliers based on the deviation of the median distribution
+        deviation = column - camera_median[:, :, np.newaxis]
+        outliers = np.logical_or(
+            deviation < self.median_range_factors[0] * camera_median[:, :, np.newaxis],
+            deviation > self.median_range_factors[1] * camera_median[:, :, np.newaxis],
+        )
+        return outliers
+
+
+class StdOutlierDetector(OutlierDetector):
+    """
+    Detect outliers based on the deviation from the camera standard deviation.
+
+    The clipping interval to set the thresholds for detecting outliers is computed by multiplying
+    the configurable factors and the camera standard deviation of the statistic values.
+    """
+
+    std_range_factors = List(
+        [-1.0, 1.0],
+        help=(
+            "Multiplicative factors (unitless) applied to the camera standard deviation"
```

Comment:
missing a space after `deviation "` and similar on the following lines.

Otherwise the docs end up like: 

>  Multiplicative factors (unitless) applied to the camera standard deviationof the provided statistic values to define a valid range based on thedeviation of the values to its camera median.

Reply:
Also it may be clearer to write this as:

> Defines the range of acceptable values (lower, upper) in units of standard deviations. Values outside of this range will be flagged as outliers.

Reply:
I think the help message needs to include that the config parameters one passed are actually factors which are multiplied by either the camera median or std. Regarding the second sentences, it's values **whose deviation to the camera median** lays outside this range are flagged as outliers.

Suggestions:
StdOutlierDetector
```python
        help=(
            "Multiplicative factors (unitless) applied to the camera standard deviation "
            "of the provided statistic values to define the range of acceptable values (lower, upper) "
            "in units of standard deviations. Values ​​whose deviation to the camera median lays outside "
            "of this range will be flagged as outliers."
        ),
```
MedianOutlierDetector
```python
        help=(
            "Multiplicative factors (unitless) applied to the camera median "
            "of the provided statistic values to define the range of acceptable values (lower, upper) "
            "in units of medians. Values ​​whose deviation to the camera median lays outside "
            "of this range will be flagged as outliers."
        ),
```

Reply:
I'd prefer Karl's formulation. It is simpler, more concise and easier to read imo.

Reply:
Yes, and for the mean, it would just be:

> Defines the range of acceptable values (lower, upper) as fractional deviations from the mean. Values outside of this range will be flagged as outliers.